tests/components/unifi/test_sensor.py

```diff
@@ -1424,3 +1424,163 @@ async def test_device_uptime(
+                                "target": "google.com",
+                                "type": "icmp",
+                            },
+                            {
+                                "availability": 100.0,
+                                "latency_average": 30,
+                                "target": "1.1.1.1",
+                                "type": "icmp",
+                            },
+                        ],
+                    },
+                    "WAN2": {
+                        "monitors": [
+                            {
+                                "availability": 0.0,
+                                "target": "www.microsoft.com",
+                                "type": "icmp",
+                            },
+                            {
+                                "availability": 0.0,
+                                "target": "google.com",
+                                "type": "icmp",
+                            },
+                            {"availability": 0.0, "target": "1.1.1.1", "type": "icmp"},
+                        ],
+                    },
+                },
+                "state": 1,
+                "type": "usw",
+                "version": "4.0.42.10433",
+            }
+        ]
+    ],
+)
+async def test_wan_monitor_latency(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    mock_websocket_message,
+    config_entry_factory: Callable[[], ConfigEntry],
+    device_payload: list[dict[str, Any]],
+) -> None:
+    """Verify that monitor latency sensors are working as expected."""
+
+    await config_entry_factory()
+
+    assert len(hass.states.async_all()) == 6
+    assert len(hass.states.async_entity_ids(SENSOR_DOMAIN)) == 2
+
+    microsoft_latency_entry = entity_registry.async_get(
+        "sensor.mock_name_microsoft_wan_latency"
+    )
+    assert microsoft_latency_entry.disabled_by == RegistryEntryDisabler.INTEGRATION
+    assert microsoft_latency_entry.entity_category is EntityCategory.DIAGNOSTIC
+
+    google_latency_entry = entity_registry.async_get(
+        "sensor.mock_name_google_wan_latency"
+    )
+    assert google_latency_entry.disabled_by == RegistryEntryDisabler.INTEGRATION
+    assert google_latency_entry.entity_category is EntityCategory.DIAGNOSTIC
+
+    cloudflare_latency_entry = entity_registry.async_get(
+        "sensor.mock_name_cloudflare_wan_latency"
+    )
+    assert cloudflare_latency_entry.disabled_by == RegistryEntryDisabler.INTEGRATION
+    assert cloudflare_latency_entry.entity_category is EntityCategory.DIAGNOSTIC
+
+    # Enable entity
+    entity_registry.async_update_entity(
+        entity_id="sensor.mock_name_microsoft_wan_latency", disabled_by=None
+    )
+    entity_registry.async_update_entity(
+        entity_id="sensor.mock_name_google_wan_latency", disabled_by=None
+    )
+    entity_registry.async_update_entity(
+        entity_id="sensor.mock_name_cloudflare_wan_latency", disabled_by=None
+    )
```

Comment:
We could deduplicate quite a lot in this test if you input entity_id, state, and the updated state in pytest parametrize, similar to the test_client_uptime

Reply:
I think I achieved it

Reply:
Looks great